ordered_product/views.py

```diff
@@ -0,0 +1,141 @@
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def search_supplier_catalog_by_user_name(request):
+    supplier_user_name = str(request.GET['supplier_user_name'])
+    message = ""
+    supplier_has_delivery_in_client_area = False
+    try:
+        user = User.objects.get(username=supplier_user_name)
+        supplier = Supplier.objects.get(supplier_account=user)
+        deliveries = DeliveryLocation.objects.filter(user_name=supplier)
+        for delivery in deliveries:
+            if delivery.location == client_check(request.user).area:
+                supplier_has_delivery_in_client_area = True
+        if not supplier_has_delivery_in_client_area:
+            raise Exception
+        supplier_products = SupplierProduct.objects.filter(user_name=supplier)
+    except Exception:
+        supplier_products = []
+        message = "No products found."
+    return render(request, 'ordered_product/new_order.html', {'supplier_products': supplier_products,
+                                                              'message': message})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def find_delivery(request, supplier_product_id):
+    try:
+        supplier_product = SupplierProduct.objects.get(supplier_product_id=int(supplier_product_id))
+    except Exception:
+        return redirect('Main Page')
+    client = client_check(request.user)
+    deliveries = DeliveryLocation.objects.filter(user_name=supplier_product.user_name, location=client.area)
+    message = ""
+    if not deliveries:
+        message = "No delivery found at '"+client.area+"'"
+    return render(request, 'ordered_product/new_order.html', {'deliveries': deliveries,
+                                                              'sup_pro_id': supplier_product_id,
+                                                              'message': message})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def order_summary(request, supplier_product_id, delivery_location_id):
+    try:
+        supplier_product = SupplierProduct.objects.get(supplier_product_id=int(supplier_product_id))
+        delivery_location = DeliveryLocation.objects.get(id=int(delivery_location_id))
+    except Exception:
+        return redirect('Main Page')
+    return render(request, 'ordered_product/new_order.html', {'delivery_location': delivery_location,
+                  'supplier_product': supplier_product})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def complete_order(request, supplier_product_id, delivery_location_id):
+    try:
+        quantity = int(request.GET['quantity'])
+        delivery_location = DeliveryLocation.objects.get(id=int(delivery_location_id))
+        supplier_product = SupplierProduct.objects.get(supplier_product_id=int(supplier_product_id))
+        OrderedProduct.order(delivery_location, client_check(request.user), supplier_product, quantity)
+    except Exception:
```

Comment:
same as above

Reply:
IN this case I want to catch ObjectDoesNotExist and illegal conversion to int.

Reply:
same as above